UsageStatsLoader.inc.php

```diff
@@ -410,18 +410,31 @@ protected function getFileTypeFromFile($file) {
 	 * @param $page string
 	 * @param $op string
 	 * @param $args array
-	 * @return array (assocId, assocType)
+	 * @return array (assocId, assocType, representationId)
 	 */
 	protected function getAssoc($assocType, $contextPaths, $page, $op, $args) {
-		$assocId = $assocTypeToReturn = null;
+		$assocId = $assocTypeToReturn = $representationId = null;
 		switch ($assocType) {
 			case ASSOC_TYPE_SUBMISSION:
+				$publicationId = null;
 				if (!isset($args[0])) break;
+				// If the operation is 'view' and the arguments count > 1
+				// the arguments must be: $submissionId/version/$publicationId.
+				// Else, it is not considered hier, as submission abstract count.
```

Comment:
Is it possible to have a `view` op with multiple arguments when looking at a file? For example, a galley view in my test install is at the following URL:

```
http://localhost:8000/publicknowledge/article/view/8/9
```

Is the first number a `path` and not an `arg`?

Reply:
That galley view URL is not logged by us, but if somehow logged it will be ignored here -- we consider it as file download (assocType = ASSOC_TYPE_SUBMISSION_FILE) and not submisison view -- that URL/function calls the download hook that we then consinder and count.
Both numbers are in $args -- the function Core::getArgs, used to get the $args returns them so.

Reply:
OK, just to make sure, though, would this not be included if someone were processing their raw Apache logs?

Reply:
You mean that galley URL? Yes, that could be -- apache logs the galley URL as well as the following download URL. So we would here, when processing the log file, ignore that galley URL -- that was one fix in this issue -- and only consider the download URL.